include/misc/mempool.h

```diff
@@ -59,7 +59,7 @@ struct sys_mem_pool_block {
 			.n_max = nmax,					\
 			.n_levels = _MPOOL_LVLS(maxsz, minsz),		\
 			.levels = _mpool_lvls_##name,			\
-			.flags = SYS_MEM_POOL_KERNEL			\
+			.flags = SYS_MEM_POOL_USER			\
```

Comment:
Is there a particular reason this needs to change?

Reply:
kernel defines mempool using K_MEM_POOL_DEFINE() that sets flags == SYS_MEM_POOL_KERNEL. 

For user mode compatible sys_mem_pool pools, a semaphore is used at the API
level since using that does not introduce latency. If flags is set to SYS_MEM_POOL_KERNEL all alloc()/free() will call irq_lock()/irq_unlock()